web/e2e-tests/drafts.test.ts

```diff
@@ -198,6 +200,15 @@ async function test_delete_draft(page: Page): Promise<void> {
     await page.click("body");
 }
 
+async function add_page_listeners(page: Page): Promise<void> {
+    await page.evaluate(
+        `let is_draft_overlay_transition_finished = false;
+        $("body").on("webkitTransitionEnd", "#draft_overlay", () => {
+            is_draft_overlay_transition_finished = true;
+        });`,
+    );
+}
```

Comment:
Can you explain how this logic is supposed to work? I don't understand what you're doing at all -- the `await`s above with `waitForFunction("is_draft_overlay_transition_finished");` first seem out of order?

Reply:
> Can you explain how this logic is supposed to work?

Sure, so we first declare a variable `is_draft_overlay_transition_finished` which will help us track if drafts overlay animation is finished or not, than we register the `webkitTransitionEnd` event with `#drafts-overlay` which will trigger whenever we open or close the drafts modal and it's transition is finished. 

> the awaits above with waitForFunction("is_draft_overlay_transition_finished"); first seem out of order?

We are registering the `webkitTransitionEnd` event when we being testing, see line number [255](https://github.com/zulip/zulip/pull/24463/files/a70d9ca1aa517b181f8fcbde886f31a09581281a#diff-f266a8a6c47295e21055fe3b589f44258ba09e71c7dd512228f040a85eeb8074R255) and after we reload the [page](https://github.com/zulip/zulip/pull/24463/files/a70d9ca1aa517b181f8fcbde886f31a09581281a#diff-f266a8a6c47295e21055fe3b589f44258ba09e71c7dd512228f040a85eeb8074R227).